### PR TITLE
clarifies security requirements wording for 3.1.1

### LINKED ISSUE #4119: Clarification about security requirements combination
Hi everyone,

Recently re-read the security requirements part of the spec which currently has this text:

> Security Requirement Objects that contain multiple schemes require that all schemes MUST be satisfied for a request to be authorized.
This enables support for scenarios where multiple query parameters or HTTP headers are required to convey security information.

> When a list of Security Requirement Objects is defined on the [OpenAPI Object](#openapi-object) or [Operation Object](#operation-object), only one of the Security Requirement Objects in the list needs to be satisfied to authorize the request.

It seems to me those two statements are contradicting each other. Let's take an example if we have security schemes A and B defined the the component section. And an Operation that references both.

Does the request need to satisfy A AND B (first statement) or A OR B (second statement) ?

#4082 doesn't seem to change anything with that as of this writing.
And it seems we've had many questions about that in the past: #3236 #1216 #771 and many more.

From reading other issues I believe what's meant here is that "first level entries are OR, nested entries are AND", for lack of better wording.

I thought that since we're close to a patch release, we might take the chance to clean this one up as well.

### PR DESCRIPTION
**Pull request by baywet**
fixes #4119

### CODE REVIEW

**Review by mikekistler — DISMISSED** [review]
I don't find the old wording confusing -- I think it is clear if you read it carefully -- but I don't object to this change.

File: versions/3.1.1.md [hunk 1]
```diff
@@ -3986,10 +3986,11 @@ flows:
...
 
-Security Requirement Objects that contain multiple schemes require that all schemes MUST be satisfied for a request to be authorized.
+A Security Requirement Object MAY refer to multiple security schemes in which case all schemes MUST be satisfied for a request to be authorized.
 This enables support for scenarios where multiple query parameters or HTTP headers are required to convey security information.
 
-When a list of Security Requirement Objects is defined on the [OpenAPI Object](#openapi-object) or [Operation Object](#operation-object), only one of the Security Requirement Objects in the list needs to be satisfied to authorize the request.
+When the security field is defined on the [OpenAPI Object](#openapi-object) or [Operation Object](#operation-object) and contains multiple Security Requirement Objects, only one of the entries in the list needs to be satisfied to authorize the request.
+This enables support for scenarios where the API supports alternative security schemes, or when they are optional.
```
Comment (kevinswiber) [inline versions/3.1.1.md:3993]: I might say
```suggestion
This enables support for scenarios where the API allows multiple, independent security schemes.

An empty Security Requirement Object (`{}`) indicates anonymous access is supported.
```

**Comment by kevinswiber** [comment]
@baywet Thank you for trying to provide additional clarity around this. It's the most complicated part of the spec, IMHO.

File: versions/3.1.1.md [hunk 2]
```diff
@@ -3986,10 +3986,11 @@ flows:
 Lists the required security schemes to execute this operation.
 The name used for each property MUST correspond to a security scheme declared in the [Security Schemes](#security-scheme-object) under the [Components Object](#components-object).
 
-Security Requirement Objects that contain multiple schemes require that all schemes MUST be satisfied for a request to be authorized.
+A Security Requirement Object MAY refer to multiple security schemes in which case all schemes MUST be satisfied for a request to be authorized.
 This enables support for scenarios where multiple query parameters or HTTP headers are required to convey security information.
 
-When a list of Security Requirement Objects is defined on the [OpenAPI Object](#openapi-object) or [Operation Object](#operation-object), only one of the Security Requirement Objects in the list needs to be satisfied to authorize the request.
+When the security field is defined on the [OpenAPI Object](#openapi-object) or [Operation Object](#operation-object) and contains multiple Security Requirement Objects, only one of the entries in the list needs to be satisfied to authorize the request.
```
Comment (ralfhandl) [inline versions/3.1.1.md:3992]:
```suggestion
When the `security` field is defined on the [OpenAPI Object](#openapi-object) or [Operation Object](#operation-object) and contains multiple Security Requirement Objects, only one of the entries in the list needs to be satisfied to authorize the request.
```

**Review by ralfhandl — APPROVED** [review]

**Review by ralfhandl — APPROVED** [review]

**Review by kevinswiber — APPROVED** [review]
LGTM!

**Review by ralfhandl — APPROVED** [review]

**Review by lornajane — APPROVED** [review]

**Review by darrelmiller — APPROVED** [review]

**Comment by baywet** [comment]
3.0.4 PR #4129

**Pull request merged**